packages/access-policy/src/__tests__/AccessPolicy.test.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Tupaia MediTrak
+ * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
+ */
+
+import { AccessPolicy } from '..';
+
+const policy = {
+  DL: ['Public'],
+  KI: ['Public', 'Admin'],
+  SB: ['Royal Australasian College of Surgeons'],
+};
+
+describe('AccessPolicy', () => {
+  const accessPolicy = new AccessPolicy(policy);
+
+  describe('allows', () => {
+    const testData = [
+      [
+        'should allow access to an entity that has at least one permission group, when no permission group is specified',
+        ['DL', ''],
+        true,
+      ],
+      ['should not allow access to an entity that has no permission groups', ['VU', ''], false],
+      [
+        'should allow access to a permission group when a matching entity is specified',
+        ['DL', 'Public'],
+        true,
+      ],
+      [
+        'should not allow access to a permission group that exists for a different entity than the one specified',
+        ['DL', 'Admin'],
+        false,
+      ],
+      ['should not allow access with no entity', ['', ''], false],
+      ['should not allow access with null', [null, ''], false],
+    ];
+
+    it.each(testData)('%s', (_, [entity, permissionGroup], expected) => {
+      expect(accessPolicy.allows(entity, permissionGroup)).toBe(expected);
+    });
+  });
+
+  describe('allowsSome', () => {
+    describe('no permission group', () => {
+      const testData = [
+        [
+          'should allow access to an entity that has at least one permission group, when no permission group is specified',
+          ['DL'],
+          true,
+        ],
+        ['should not allow access to an entity that has no permission groups', ['VU'], false],
+        ['should not allow access with no entity', [], false],
```

Comment:
We also need one test case for `undefined`

```suggestion
        ['should not allow access with no entity', [], false],
        ['should not allow access with no entity', undefined, false],
```

This corresponds to the following test case in the existing code:

```js
expect(accessPolicy.allowsSome()).to.equal(false);
```